### PR TITLE
Monster jump

### PR DESCRIPTION
**Pull request by Verzijl**
The monsters/walkers are now jumping randomly around the map.

### CODE REVIEW

**Comment by jimhommes** [comment]
Please write some tests as the coverage of Walker.java went from 75% to 66%. 

If you do that, :shipit:

File: src/test/java/model/WalkerTest.java [hunk 1]
```diff
@@ -54,10 +54,10 @@ public void testSwitchingDirectionTrue() {
 	@Test
 	public void testMove() throws Exception {
 		walker.move();
-		assertEquals(Settings.MONSTER_SPEED , walker.getX(), 0);
+		assertEquals(37.0 , walker.getX(), 0);
```
Comment (toinehartman) [inline src/test/java/model/WalkerTest.java]: Why would the x coordinate be different now? Jumping is only vertical and looking at the code, it does not influence speed in the x direction. This probably means you broke something, adjusting the test is not always the right solution if it fails.

File: src/main/java/model/Walker.java [hunk 2]
```diff
@@ -54,19 +110,55 @@ public void move() {
...
+            setX(walkerMaxX - getWidth());
+        }
+
+        if (getY() < walkerMinY) {
+            setY(walkerMinY);
+        } else if (getY() + getHeight() > walkerMaxY) {
+            setY(walkerMaxY - getHeight());
+        }
+    }
+
```
Comment (toinehartman) [inline src/main/java/model/Walker.java]: This looks like the code is inspired by the jumping of the player. This makes a lot of duplicate code, which should be put in GravityObject.

File: src/main/java/model/Walker.java [hunk 3]
```diff
@@ -96,4 +188,14 @@ private void moveHorizontal() {
     public void switchDirection() {
         setFacingRight(!isFacingRight());
     }
+
+    private int randInt() {
+        Random rand = new Random();
+        int min = 1;
+        int max = 200;
+        int res = rand.nextInt((max - min) + 1) + min;
+        System.out.println(res);
```
Comment (toinehartman) [inline src/main/java/model/Walker.java]: Please take out prints used for debugging. If you really need to display this information, use the Logger and a String that makes clear what this number means.

File: src/main/java/model/Walker.java [hunk 4]
```diff
@@ -46,6 +91,17 @@ public Walker(double x,
...
     public void move() {
         if (!isCaughtByBubble()) {
+
+            moveCollisionChecker();
+
+            if (jumpCounter < 12) {
+                jumpCounter++;
+            }
+
+            if (jumpCounter == 12) {
```
Comment (toinehartman) [inline src/main/java/model/Walker.java]: This condition can never be true if the first one is, so use an `else if`.

**Comment by JPluim** [comment]
When monsters are walking in the top of the screen, it looks like they can jump pretty high. It doesn't hurt, but will add as an issue for further improvements. :shipit:

**Pull request merged**